Core/Python/add_device_to_static_group.py

```diff
@@ -20,33 +20,158 @@
+                next_link_url = "https://%s" + data['@odata.nextLink']
+            if device_data is None:
+                device_data = data["value"]
+            else:
+                device_data += data["value"]
+        else:
+            print("Unable to retrieve device list from %s" % ome_ip_address)
+            exit(1)
+
+    return device_data
+
+
+def get_group_id_by_name(ome_ip_address: str, group_name: str, headers: dict) -> int:
+    """
+    Retrieves the ID of a group given its name.
+
+    Args:
+        ome_ip_address: The IP address of the OME server
+        group_name: The name of the group whose ID you want to resolve.
+        headers: Headers used for authentication to the OME server
+
+    Returns: Returns the ID of the group as an integer.
+
+    """
+
+    print("Searching for the requested group.")
+    groups_url = "https://%s/api/GroupService/Groups?$filter=Name eq '%s'" % (ome_ip_address, group_name)
+
+    group_response = requests.get(groups_url, headers=headers, verify=False)
+
+    if group_response.status_code == 200:
+        json_data = json.loads(group_response.content)
+
+        if json_data['@odata.count'] > 1:
+            print("WARNING: We found more than one name that matched the group name: " + group_name +
+                  ". We are picking the first entry.")
+        if json_data['@odata.count'] == 1 or json_data['@odata.count'] > 1:
+            group_id = json_data['value'][0]['Id']
+            if not isinstance(group_id, int):
+                print("The server did not return an integer ID. Something went wrong.")
+                exit(1)
+            return group_id
+        else:
+            print("Error: We could not find the group " + group_name + ". Exiting.")
+            exit(1)
+    else:
+        print("Unable to retrieve groups. Exiting.")
+        exit(1)
+
+
+def get_device_id_by_name(ome_ip_address: str, device_name: str, headers: dict) -> int:
+    """
+    Resolves the name of a server to an OME ID
+
+    Args:
+        ome_ip_address: IP address of the OME server
+        device_name: Name of the device whose ID you want to resolve
+        headers: Headers used for authentication to the OME server
+
+    Returns:
+        The ID of the server or 0 if it couldn't find it
+    """
+
+    url = "https://%s/api/DeviceService/Devices?$filter=DeviceName eq \'%s\'" % (ome_ip_address, device_name)
+
+    response = requests.get(url, headers=headers, verify=False)
+    print("Getting the device ID for system with name " + device_name + "...")
+
+    if response.status_code == 200:
+        json_data = response.json()
+
+        if json_data['@odata.count'] > 1:
+            print("WARNING: We found more than one name that matched the device name: " + device_name +
+                  ". We are skipping this entry.")
+        elif json_data['@odata.count'] == 1:
+            server_id = json_data['value'][0]['Id']
+            if not isinstance(server_id, int):
```

Comment:
Do you see these errors? I'd think this always returns an integer value that is greater than 0 or the filter call wouldn't have returned any results.

Reply:
No, but I accounted for it since the spec didn't say it couldn't happen.